File: catkin_ws/src/.old_archive/learning_topic/src/person_subscriber.cpp
// subscribe the topic: /person_info
// user defined message type: learning_topic::Person

#include <ros/ros.h>
#include "learning_topic/Person.h"

void personInfoCallback(const learning_topic::Person::ConstPtr& msg){
	// print message received
	ROS_INFO("Subscribe person info: Name:%s  age:%d  sex:%d",
			msg->name.c_str(), msg->age, msg->sex);
}

int main (int argc, char **argv)
{
	// node init. Node name: velocity_publisher
	ros::init(argc, argv, "person_subscriber");
	
	// create node handle
	ros::NodeHandle n;
	
	// Create subscriber. 
	// Subscribed Topic name: /person_info
	// register call back function
	ros::Subscriber person_info_sub = n.subscribe("/person_info",10,personInfoCallback);
	
	// loop, waiting for call back function. If message coming in, call poseCallback
	ros::spin();
	
	return 0;
	
}

File: catkin_ws/src/.old_archive/learning_topic/script/person_subscriber.py
# !/usr/bin/env python3
# -*- coding: utf-8 -*-
# subscribe the topic /turtle1/pose
# msg type: turtlesim:Pose

import rospy
from learning_topic.msg import Person

def personInfoCallback(msg):
	rospy.loginfo("Subscribe person info: name:%s age:%d sex:%d",msg.name, msg.age, msg.sex)
	
def pose_subscriber():
	# node init
	rospy.init_node('person_subscriber',anonymous=True)
	
	# Create subscriber. 
	# Subscribed Topic name: /turtle1/pose
	# register call back function
	rospy.Subscriber('/person_info',Person, personInfoCallback)
	
	# set loop freq
	rospy.spin()
			
if __name__=='__main__':
	person_subscriber()
